### PR TITLE
BFD-3492: Simplify pipeline configuration

### PR DESCRIPTION
**Pull request by mjburling**
**JIRA Ticket:**
BFD-3492

### What Does This PR Do?
This PR hopes to simplify the pipeline configuration strategy through deduplication and nudging BFD toward further standardization. This continues to shift the configuration, where appropriate, to the _right_: where sufficient context can be provided to the workload, pipeline workloads self-configure by fetching configuration parameters from relevant hierarchies encoded in AWS SSM directly, instead of relying on intermediate steps to fetch their configuration.

Short of fully standardizing the configurations, I have sought to include, at least, *all* configuration keys at issue. Checking for pipeline-related configuration with this change set can be done using bash:

```sh
for e in prod prod-sbx test; do
    printf "$e: "
    yq '.. |  key' "ops/terraform/services/base/values/${e}.yaml" | grep pipeline | wc -l
done
```
Yields the following result:
```
prod: 58
prod-sbx: 58
test: 58
```

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

### What Security Implications Does This PR Have?

Please indicate if this PR does any of the following:  

* Adds any new software dependencies
* Modifies any security controls
* Adds new transmission or storage of data
* Any other changes that could possibly affect security? 

* [x] I have considered the above security implications as it relates to this PR. (If one or more of the above apply, it cannot be merged without the ISSO or team security engineer's (`@sb-benohe`) approval.) 

### Validation

* Ephemeral environment support:
  * I applied these changes to the `test` environment under the `base` terraservice before creating a partial 3492-test environment
  * I created a SNAPSHOT-based AMI image from the changes here for evaluation and successfully leveraged this AMI in 3492
  * 3492-test's pipeline instances exhibited the expected behavior
* Established environment support for `test`, `prod-sbx`, and `prod`:
  * Applied changes from this branch to the `base` terraservice for all three established environments
  * Applied changes from this branch to the `pipeline` terraservice for all three established environments
  * Observed expected functionality of the pipeline instances for RDA and CCW workloads for all three established environments

### CODE REVIEW

File: ops/ansible/roles/bfd-pipeline/templates/bfd-pipeline-service.sh.j2 [hunk 1]
```diff
@@ -1,73 +1,27 @@
...
+
+export BFD_DB_URL='{{ db_url }}' {# derived from terraform control plane #}{{ '\n' }}
+{{ '#' if db_password_override|default('', true)|length<1 else '' }}export BFD_DB_PASSWORD='{{ db_password_override|default('', true) }}'                      {# application defaults to ssm db/password #}{{ '\n' }}
+{{ '#' if db_username_override|default('', true)|length<1 else '' }}export BFD_DB_USERNAME='{{ db_username_override|default('', true) }}'                      {# application defaults to ssm db/username #}{{ '\n' }}
+{{ '#' if db_auth_type_override|default('', true)|length<1 else '' }}export BFD_DB_AUTH_TYPE='{{ db_auth_type_override|default('', true) }}'                   {# application defaults to ssm db/auth_type #}{{ '\n' }}
+{{ '#' if db_max_connections_override|default('', true)|length<1 else '' }}export BFD_DB_MAX_CONNECTIONS='{{ db_max_connections_override|default('', true) }}' {# application defaults to ssm db/max_connections #}{{ '\n' }}
+
+export BFD_NEW_RELIC_APP_NAME="BFD Pipeline{{ pipeline_instance_type if pipeline_instance_type == ' ccw' else '' }}"
```
[the remark this answers is not in the document]
Comment (mjburling) [reply ops/ansible/roles/bfd-pipeline/templates/bfd-pipeline-service.sh.j2]: > `' ccw'` is the space at the beginning a misake?

Yes. The implementation has changed here. Thanks for pointing this out!

**Review by timothy-ellis-ky — APPROVED** [review]
understood as explained

File: ops/ansible/roles/bfd-pipeline/templates/bfd-pipeline-service.sh.j2 [hunk 2]
```diff
@@ -1,73 +1,27 @@
...
+export BFD_RDA_JOB_ENABLED={{ enabled if pipeline_instance_type == 'rda' else 'false' }} {# derived from terraform control plane #}{{ '\n' }}
+
+export BFD_DB_URL='{{ db_url }}' {# derived from terraform control plane #}{{ '\n' }}
+{{ '#' if db_password_override|default('', true)|length<1 else '' }}export BFD_DB_PASSWORD='{{ db_password_override|default('', true) }}'                      {# application defaults to ssm db/password #}{{ '\n' }}
+{{ '#' if db_username_override|default('', true)|length<1 else '' }}export BFD_DB_USERNAME='{{ db_username_override|default('', true) }}'                      {# application defaults to ssm db/username #}{{ '\n' }}
+{{ '#' if db_auth_type_override|default('', true)|length<1 else '' }}export BFD_DB_AUTH_TYPE='{{ db_auth_type_override|default('', true) }}'                   {# application defaults to ssm db/auth_type #}{{ '\n' }}
+{{ '#' if db_max_connections_override|default('', true)|length<1 else '' }}export BFD_DB_MAX_CONNECTIONS='{{ db_max_connections_override|default('', true) }}' {# application defaults to ssm db/max_connections #}{{ '\n' }}
```
Comment (malessi) [inline ops/ansible/roles/bfd-pipeline/templates/bfd-pipeline-service.sh.j2:10]: Changes introduced in BFD-3507 have updated the corresponding parameter from `db/max_connections` to `db/hikari/max_pool_size`:

```suggestion
{{ '#' if db_max_connections_override|default('', true)|length<1 else '' }}export BFD_DB_HIKARI_MAX_POOL_SIZE='{{ db_hikari_max_pool_size_override|default('', true) }}' {# application defaults to ssm db/hikari/max_pool_size #}{{ '\n' }}
```

**Review by malessi — CHANGES_REQUESTED** [review]
One _tiny_ SSM parameter update, and then this is good to go!

**Review by malessi — APPROVED** [review]
Everything looks good!

**Review by timothy-ellis-ky — APPROVED** [review]
updates approved

**Pull request merged**